contactaccessrights/contactaccessrights.php

```diff
@@ -190,3 +196,46 @@ function contactaccessrights_civicrm_pageRun($page) {
     CRM_Core_Resources::singleton()->addScriptFile($extName, 'js/dist/access-rights.min.js', 1010);
   }
 }
+
+/**
+ * Implementation of hook_addContactMenuActions to add the
+ * Contact Access menu group to the contact actions menu.
+ *
+ * @param ActionsMenu $menu
+ *
+ * @throws \Exception
+ */
+function contactaccessrights_addContactMenuActions(ActionsMenu $menu) {
+  $contactID = empty($_GET['cid']) ? '' : $_GET['cid'];
+  if (!$contactID) {
+    return;
+  }
+
+  $contactUserInfo = ContactHelper::getUserInformation($contactID);
+
+  if (empty($contactUserInfo['cmsId'])) {
+    return;
+  }
+
+  $userPermissions = CMSUserPermissionsFactory::create();
+
+  if (!$userPermissions->check($contactUserInfo, ['access CiviCRM'])) {
```

Comment:
Tbh, I was expecting this check to be inside the `CRM_Contactaccessrights_Helper_ContactActionsMenu_ContactAccessActionGroup` class, as it already has some other permissions checks. I think it will the easier to understand things if all the permissions checks are in one single place

Reply:
good Idea.